internal/connector/pingone/platform/resources/pingone_notification_template_content.go

```diff
@@ -0,0 +1,124 @@
+package resources
+
+import (
+	"fmt"
+
+	"github.com/patrickcping/pingone-go-sdk-v2/management"
+	"github.com/pingidentity/pingctl/internal/connector"
+	"github.com/pingidentity/pingctl/internal/connector/common"
+	"github.com/pingidentity/pingctl/internal/logger"
+)
+
+// Verify that the resource satisfies the exportable resource interface
+var (
+	_ connector.ExportableResource = &PingoneNotificationTemplateContentResource{}
+)
+
+type PingoneNotificationTemplateContentResource struct {
+	clientInfo *connector.SDKClientInfo
+}
+
+// Utility method for creating a PingoneNotificationTemplateContentResource
+func NotificationTemplateContent(clientInfo *connector.SDKClientInfo) *PingoneNotificationTemplateContentResource {
+	return &PingoneNotificationTemplateContentResource{
+		clientInfo: clientInfo,
+	}
+}
+
+func (r *PingoneNotificationTemplateContentResource) ExportAll() (*[]connector.ImportBlock, error) {
+	l := logger.Get()
+
+	l.Debug().Msgf("Fetching all %s resources...", r.ResourceType())
+
+	importBlocks := []connector.ImportBlock{}
+
+	l.Debug().Msgf("Generating Import Blocks for all %s resources...", r.ResourceType())
+
+	// This is weird... the provider mentions many possible template types,
+	// but pingone console and the API only support the following types:
+	validTemplateNames := []management.EnumTemplateName{
```

Comment:
@patrickcping Any insight into why any other temple type than defined here returns a 500 server code when supplied to the `ManagementAPIClient.NotificationsTemplatesApi.ReadAllTemplateContents`? The Pingone provider documentation mentions a total of 16 `template_name` options, though I was only able to get these 8 working here.